
syntax = "proto3";

enum LoaderStatus {
    Ok = 0;
    False = 1;
}

message CreateDataloaderRequest {
    uint32 dataset_id = 1;
    repeated uint32 keys = 2;
}

message CreateDataloaderResponse {
    string shared_mem_file = 2;
    uint64 loader_id = 3;
}


message NextRequest {
    uint64 loader_id = 1;
}

message NextResponse {
    uint64 address = 2;
}

message DeleteDataloaderRequest {
    uint64 loader_id = 3;
}

message DeleteDataloaderResponse {
    LoaderStatus resp = 3;
}

service DataLoader {
    rpc CreateDataloader(CreateDataloaderRequest) returns (CreateDataloaderResponse);
    rpc Next(NextRequest) returns (NextResponse);
    rpc DeleteDataloader(DeleteDataloaderRequest) returns (DeleteDataloaderResponse);
}